### DISCUSSION TITLE
Referencing a kcl.mod project

### DISCUSSION THREAD

**Discussion by nkabir** [kcl-lang/kcl#620] [General]
I'd like to use kcl file in one project by referencing a kcl.mod folder in another project. Is this possible?

```
project-a/
  myconfig.k
project-b/
  configschema/
    kcl.mod
    configfile1.k
    configfile2.k
```

I tried setting an environment variable `KCL_MOD` but that did not work. It would be helpful to manage a kcl project in one folder and reference it from other projects. Is this possible?

**Comment by Peefy** [comment]
I guess what you mean is to import projectB files from projectA, right? One way is to maintain the same `kcl.mod` file as shown below, and the other way is to use kpm to manage your dependencies

+ Method 1

```shell
kcl.mod
projecta/
  myconfig.k
projectb/
  configschema/
    configfile1.k
    configfile2.k
```

The code of `myconfig.k` maybe

```python
import projectb.configschema

foo = configschema.Foo {}
```

+ Method2 (Example: https://github.com/kcl-lang/examples/tree/main/package-management/git/my_package)

```shell
# You need to complete the maintenance of projectB in advance, such as a separate git repo
kpm init projectA && kpm add projectB && kpm run
```

**Comment by nkabir** [comment]
Thank you. That works!

However, I'd like to use kcl files everywhere because they're very helpful. It would be great if, instead of placing `kcl.mod` files everywhere, it would be possible to set a KCL_MOD environment variable like PATH and have kcl search those locations.

**Reply by Peefy** [reply]
Ping @zong-zhe Considering this feature for kpm like jsonnet package manager.

**Reply by zong-zhe** [reply]
Hi @nkabir, kcl cli supports describing dependeny location with `-E`.

The command is used as follows：
```
kcl main.k -E <package_name>=<package_local_real_path>
```

In `main.k`, you can import the kcl package by:
```
import <package_name>.sub1.sub2
```

I have created a demo according to your description, you can try if it is helpful.

1. The program in `project-a/myconfig.k`.

```
import configschema

myconfig = configschema.other_config

```

2. The program in `project-b/configschema/main.k`.
```
other_config = “The config in configschema”
```

3. Compile with the following command.
```
$ kcl project-a/myconfig.k -E configschema=project-b/configschema/
```

![image](https://github.com/kcl-lang/kcl/assets/68977949/78955c7c-2bdf-425c-9c1d-ef43fc0b79d1)

If there are a lot of kcl dependencies, this can make the kcl build command very long, and we are also working on this problem by managing the packages of the local path through kcl's package manager [kpm](https://github.com/kcl-lang/kpm), you can track the progress in https://github.com/kcl-lang/kpm/issues/133.

**Comment by nkabir** [comment]
That will work. Thank you. KCL is fantastic.
